src/lib/drivers/device/Device.hpp

```diff
@@ -114,6 +114,23 @@ class __EXPORT Device
 	 */
 	virtual int	write(unsigned address, void *data, unsigned count) { return -ENODEV; }
 
+	/**
+	 * Read a register from the device.
+	 *
+	 * @param		The register to read.
+	 * @return		The value that was read.
+	 */
+	virtual uint8_t read_reg(unsigned reg) { return -ENODEV; }
+
+	/**
+	 * Write a register in the device.
+	 *
+	 * @param reg		The register to write.
+	 * @param value		The new value to write.
+	 * @return		OK on success, negative errno otherwise.
+	 */
+	virtual int write_reg(unsigned reg, uint8_t value) { return -ENODEV; }
```

Comment:
```suggestion
	virtual int write_reg8(unsigned reg, uint8_t value) { return -ENODEV; }
```
And can you implement these for NuttX too?

Reply:
No, as stated in comments above.